app-creator/src/widgets/ui-map/ui-map.ts

```diff
@@ -181,11 +263,30 @@ export class Map extends LitElement {
         this.apiKey = value;
         break;
     }
-    this.initMap();
+    this.updateMap();
+  }
+
+  getCustomMapStyle(value: string): google.maps.MapTypeStyle[] {
+    switch (value) {
```

Comment:
Should we also have a `custom` case, where users can input their own style JSON?

Reply:
The custom case is currently handled by a separate custom map styles input. Anything that is put in there will override what's in the select. 

```Typescript
case 'mapStyles':
        this.mapStyles = this.getCustomMapStyle(value);
        break;
case 'customMapStyles':
        this.customMapStyles = value;
        break;
```

Some other approaches we can take:
1. Add custom as part of the select menu, and display/hide the custom input whether we select select or not.
2. Disable the select input when there is data in the custom style input to show the user that it is being overwritten.